scripts/ci/opencode_coverage_identity.py

```
@@ -0,0 +1,199 @@
+    raw = sys.stdin.read() if not path or path == "-" else Path(path).read_text(encoding="utf-8")
+    loaded = json.loads(raw)
+    if isinstance(loaded, Mapping) and isinstance(loaded.get("check_runs"), list):
+        loaded = loaded["check_runs"]
+    if not isinstance(loaded, list):
+        raise CoverageQuoteError("coverage identity payload must be a check-run array")
+    return [item for item in loaded if isinstance(item, Mapping)]
+
+
+def fetch_check_runs(repo: str, head_sha: str) -> list[Mapping[str, Any]]:
+    """Read exact-head check-runs through gh without invoking a shell."""
+    completed = subprocess.run(
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)